app/models/validations/financial_validations.rb

```diff
@@ -181,20 +181,34 @@ def validate_rent_range(record)
     return if record.startdate.blank?
 
     collection_year = record.collection_start_year
-    rent_range = LaRentRange.find_by(start_year: collection_year, la: record.la, beds: record.beds, lettype: record.lettype)
+
+    max_beds_in_rent_ranges = 4
+    validation_beds = if record.needstype == 2
+                        0
+                      else
+                        record.beds.nil? ? nil : [record.beds, max_beds_in_rent_ranges].min
+                      end
+
+    rent_range = LaRentRange.find_by(start_year: collection_year, la: record.la, beds: validation_beds, lettype: record.lettype)
 
     if rent_range.present? && !weekly_value_in_range(record, "brent", rent_range.hard_min, rent_range.hard_max) && record.brent.present? && record.period.present?
       if record.weekly_value(record["brent"]) < rent_range.hard_min
         record.errors.add :brent, I18n.t("validations.financial.brent.below_hard_min")
         record.errors.add :beds, I18n.t("validations.financial.brent.beds.below_hard_min")
         record.errors.add :la, I18n.t("validations.financial.brent.la.below_hard_min")
+        record.errors.add :postcode_known, I18n.t("validations.financial.brent.postcode_known.below_hard_min")
+        record.errors.add :scheme_id, I18n.t("validations.financial.brent.scheme_id.below_hard_min")
+        record.errors.add :location_id, I18n.t("validations.financial.brent.location_id.below_hard_min")
         record.errors.add :rent_type, I18n.t("validations.financial.brent.rent_type.below_hard_min")
         record.errors.add :needstype, I18n.t("validations.financial.brent.needstype.below_hard_min")
         record.errors.add :period, I18n.t("validations.financial.brent.period.below_hard_min")
-      else
+      elsif !(record.beds.present? && record.beds > max_beds_in_rent_ranges)
```

Comment:
```suggestion
      elsif record.beds.blank? || record.beds < LaRentRange::MAX_BEDS
```

I think this might be better on the `LaRentRange` model itself, otherwise you have a magic `4` repeated twice in the codebase. We can also flip the logic to make it more readable. What do you think?

Reply:
Yep much nicer, wasn't sure where to put this so ended up duplicating.